sn_transfers/src/cashnotes/spend_reason.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright 2024 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use bls::{Ciphertext, PublicKey, SecretKey};
+use lazy_static::lazy_static;
+use serde::{Deserialize, Serialize};
+use xor_name::XorName;
+
+use crate::{DerivationIndex, Hash, Result, TransferError};
+
+/// The attached metadata or reason for which a Spend was spent
+#[derive(Default, Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub enum SpendReason {
+    #[default]
+    None,
+    /// Reference to network data
+    NetworkData(XorName),
+    /// Custom field for any application data
+    Custom(#[serde(with = "serde_bytes")] [u8; 64]),
```

Comment:
better use const instead of maject number of `64`,
say `MAX_CIPHER_SIZE / 2` ?

Reply:
Yeah const is good. Maybe we should not link this to the beta rewards consts, I'll make a new one